hooks/tk-multi-shotgunpanel/tk-maya_actions.py

```diff
@@ -124,8 +133,18 @@ def _create_texture_node(self, path, sg_publish_data):
         file_node = mel.eval('createRenderNodeCB -as2DTexture "" file ""')
 
         has_frame_spec, path = self._find_first_frame(path)
+
         # use the first frame instead of %04d, else maya errors out with "File Doesn't exist".
         cmds.setAttr("%s.fileTextureName" % file_node, path, type="string")
+
+        # Fetching the map type name from the file name at the path
+        entity_name = sg_publish_data["entity"]["name"]
+        pub_name = sg_publish_data["name"]
+        file_node_name = "{0}{1}".format(entity_name, pub_name.split(entity_name)[-1])
+        # Renaming the file node created to the map being loaded
+        # Overriding the file_node variable to return the newly assigned name
+        file_node = cmds.rename(file_node, file_node_name)
+
```

Comment:
Maybe this code block could be moved to a function (eg. `rename_file_node(file_node, sg_publish_data)`), because it is repeated everywhere?

Reply:
Addressed the changes suggested.